### DISCUSSION TITLE
Flutter: rpc is returning incorrect data

### DISCUSSION THREAD

**Discussion by ScottishRoss** [supabase/supabase#11510] [Questions]
Hi all,

This is a very confusing issue that I can't seem to wrap my head around. I have a database function that returns the result of this SQL Query:

```
select
  requested_to,
  requested_by,
  s.first_name,
  avatar_path,
  is_disabled
from
  public.users u
left join public.synced_users s on u.email = s.requested_by
where 
  requested_to = param_request_email and
  is_disabled = false
```

With a text parameter of an email address being passed through to it. This query, when tested in the SQL Editor, returns the two rows with the data I expect.

| requested_to            | requested_by            | first_name | avatar_path                     | is_disabled |
| ----------------------- | ----------------------- | ---------- | ------------------------------- | ----------- |
| test@gmail.com | test2@gmail.com    | Seren      | avatars/test2@gmail.com | false       |
| test@gmail.com | test3@gmail.com | Peter      | avatars/test3@gmail.com | false       |

When I call the rpc function via Flutter like so, I get completely different data:

```
  Future<dynamic> getUsersCircle(String email) async {
    final dynamic data =
        await supabase.rpc('get_user_circle', params: {'param_request_email ': email});

    return data;
  }
```

The result that's returning is this JSON string, repeated 6 times:

`{requested_to: test@gmail.com, requested_by: test2@gmail.com, first_name: Seren, avatar_path: avatars/test2@gmail.com, is_disabled: false}`

This makes no sense to me and I cannot fathom what's going on. Does anyone have any idea what I've done wrong? I am expecting two responses with different data, not the same one repeated 6 times.

Note: for now RLS is turned off on both tables.

**Comment by silentworks** [comment]
What happens when you call the function inside the SQL editor?
```sql
select get_user_circle('test@gmail.com');
```

**Reply by silentworks** [reply]
Please reply in this thread. A function will return as many records as your query is requesting to be returned.

**Reply by ScottishRoss** [reply]
Apologies, it's been a busy few weeks. Thanks for trying to help.

As noted below, it returns this weird string `(test@gmail.com,test2@gmail.com,Seren,avatars/test2@gmail.com,f)`

So it is only returning one record like the above, but for some reason, it's repeated 6 times and I can't figure out why. I can only assume setting a database function to return a record will only return a max of 1 record.

**Comment by ScottishRoss** [comment]
Sorry, I didn't see your comment until now. It returns a weird string like this:

`(test@gmail.com,test2@gmail.com,Seren,avatars/test2@gmail.com,f)`

But it should be returning two... Can database functions only return 1 record?

**Reply by steve-chavez** [reply]
Make sure your function is declared with `RETURNS SETOF ..`, otherwise you'll only get 1 row.

**Reply by ScottishRoss** [reply]
I did this as a SQL query, not as a postgres function.
